processor/processor.go

```diff
@@ -1171,14 +1190,32 @@ func (proc *Handle) updateMetricMaps(
 	}
 }
 
-func (proc *Handle) getFailedEventJobs(response transformer.Response, commonMetaData *transformer.Metadata, eventsByMessageID map[string]types.SingularEventWithReceivedAt, stage string, transformationEnabled, trackingPlanEnabled bool) ([]*jobsdb.JobT, []*types.PUReportedMetric, map[string]int64) {
-	failedMetrics := make([]*types.PUReportedMetric, 0)
+func (proc *Handle) getNonSuccessfulMetrics(response transformer.Response, commonMetaData *transformer.Metadata, eventsByMessageID map[string]types.SingularEventWithReceivedAt, stage string, transformationEnabled, trackingPlanEnabled bool) *NonSuccessfulTransformationMetrics {
+	m := &NonSuccessfulTransformationMetrics{}
+
+	var filtered, failed []transformer.TransformerResponse
+	for _, event := range response.FailedEvents {
+		if event.StatusCode == types.FilterEventCode {
```

Comment:
Shouldn't the filtered event code be returned by the user transformer as well in case the event is dropped?
IMO to be confident we need integration tests, with an actual transformer, reporting enabled (but not sending reports to a remote server) to verify that filtered events are being recorded in reporting in various stages of the processor's pipeline:
- 1 scenario for message filter
- 1 scenario for destination transformation
- 1 scenario for user transformation
- 1 scenario for tracking plan(?)

cc @pChondros